environment-deprecated.yml

```diff
@@ -6,9 +6,9 @@ channels:
 dependencies:
   - lightgbm <4.0
   - numpy
-  - python>=3.8
+  - python >=3.8
   - pre-commit
   - pandas
-  - scikit-learn <1.3.0
-  - pytest>=7.0
+  - scikit-learn >=1.3.0,<1.4
```

Comment:
It would be great if `slim-trees` could at least support `scikit-learn >=1.1`.

Maybe you could proceed as follows?
* Save the `scikit-learn` version at compression time.
* Use `packaging.version.Version` to construct if-else statements to include/exclude `missing_go_to_left`.

Reply:
> Save the scikit-learn version at compression time.

I don't think we should support using differen sklearn versions than what you used at compression time. Sklearn itself also prints out a warning if you use a different version at runtime.

If-else statements to include/exclude `missing_go_to_left` are a good idea, I implemented it 👍🏻